src/pybind/mgr/stats/fs/perf_stats.py

```diff
@@ -164,6 +170,44 @@ def notify(self, cmdtag):
             self.log.debug("client_metadata={0}, to_purge={1}".format(
                 self.client_metadata['metadata'], self.client_metadata['to_purge']))
 
+    def notify_fsmap(self):
+        #Reregister the user queries when there is a new rank0 mds
+        gid_state = FSPerfStats.get_rank0_mds_gid_state(self.module.get('fs_map'))
+        if not gid_state:
+            return
+        rank0_gid, state = gid_state
+        if (rank0_gid and (rank0_gid != self.prev_rank0_gid)):
+            if (state != 'up:active'):
+                return
+            #ronk0 MDS is up:active
+            ua_timestamp = datetime.now()
+            #Clear and wait for the empty metrics event before reregistering queries.
+            self.mx_event.clear()
+            self.mx_event.wait()
```

Comment:
Its not a  good idea to block during notify. Ideally, you should kick off the timer task when you see a new rank 0 mds.

In the timer task, check the last_updated timestamp with the one recorded when the new rank 0 mds was seen and then force register. If the last_updated timestamp is not refreshed, schedule the timer task again for the next second.